src/native/eventpipe/ep.c

```diff
@@ -302,6 +305,7 @@ void
 ep_provider_callback_data_free (EventPipeProviderCallbackData *provider_callback_data)
 {
 	ep_return_void_if_nok (provider_callback_data != NULL);
+	ep_rt_utf8_string_free (provider_callback_data->filter_data);
```

Comment:
Since we can just do a stack based version of this struct using init/fini, you should move the ep_rt_utf8_string_free (provider_callback_data->filter_data); and call ep_provider_callback_data_fini from the free method.